orchagent/orch.cpp

```diff
@@ -194,14 +194,25 @@ size_t Consumer::refillToSync()
         } while (update_size != 0);
         return total_size;
     }
-    else
+    string tableName = getTableName();
+    auto consumerTable = dynamic_cast<ConsumerTableBase *>(getSelectable());
+    if (consumerTable != NULL)
     {
         // consumerTable is either ConsumerStateTable or ConsumerTable
-        auto db = getDbConnector();
-        string tableName = getTableName();
+        auto db = consumerTable->getDbConnector();
+        auto table = Table(db, tableName);
+        return refillToSync(&table);
+    }
+    auto zmqTable = dynamic_cast<ZmqConsumerStateTable *>(getSelectable());
+    if (zmqTable != NULL)
+    {
+        auto db = zmqTable->getDbConnector();
         auto table = Table(db, tableName);
         return refillToSync(&table);
     }
+    DBConnector db("APPL_DB", 0);
+    auto table = Table(&db, tableName);
+    return refillToSync(&table);
```

Comment:
Why need this block? What is the class of `getSelectable()`? APPL_DB seems a magic choice.

Reply:
This is the default case if none of the above condition matches. Today, orchagent reads the entries from APPL DB after warm boot. I hear dash is different, but using APPL DB shouldn't be completely unreasonable.

Do you suggest to remove this default case handling? I can do that.

Reply:
I do not get "the class of getSelectable()?" in this case. If you do not either, let's remove this block.

Reply:
Done